src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,37 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle)obj;
+            return ((color == null && rectangle.color == null)
+                        || (color != null && rectangle.color != null)
+                        && rectangle.getWidth().equals(getWidth()))
```

Comment:
now I agree with the logic - good job!

but this is a single logic block where you compare the same field of two rectangles
you take `color` everywhere but suddenly `width` in the end

same for other fields

Reply:
done